libraries/abstractions/common_io/test/test_iot_tsensor.c

```diff
@@ -79,6 +81,16 @@ TEST_TEAR_DOWN( TEST_IOT_TSENSOR )
  */
 TEST_GROUP_RUNNER( TEST_IOT_TSENSOR )
 {
+    if( xtestIotTsensorMinSemaphore == NULL )
+    {
+        xtestIotTsensorMinSemaphore = xSemaphoreCreateBinary();
```

Comment:
In general, you should always check return codes here since [`xSemaphoreCreateBinary()`](https://freertos.org/xSemaphoreCreateBinary.html) can return `NULL`.

Reply:
It makes sense. I moved the block out to the test setup phase and added asserts to check against NULL return.